src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -894,5 +894,61 @@ private static VariableDefinition GetLocalDef (Instruction operation, Collection
 			Instruction operation,
 			ValueNodeList methodParams,
 			out ValueNode methodReturnValue);
+
+		private void ScanStelem (
+			Instruction operation,
+			Stack<StackSlot> currentStack,
+			MethodBody methodBody,
+			int curBasicBlock)
+		{
+			StackSlot valueToStore = PopUnknown (currentStack, 1, methodBody, operation.Offset);
+			StackSlot indexToStoreAt = PopUnknown (currentStack, 1, methodBody, operation.Offset);
+			StackSlot arrayToStoreIn = PopUnknown (currentStack, 1, methodBody, operation.Offset);
+			int? indexToStoreAtInt = indexToStoreAt.Value.AsConstInt ();
+			foreach (var array in arrayToStoreIn.Value.UniqueValues ()) {
+				if (array is ArrayValue arrValue) {
+					if (indexToStoreAtInt == null) {
+						// Since we can't know the current index we're storing the value at, clear all indices.
+						// That way we won't accidentally think we know the value at a given index when we cannot.
+						foreach (var knownIndex in arrValue.IndexValues.Keys) {
+							StoreMethodLocalValue (arrValue.IndexValues, UnknownValue.Instance, knownIndex, curBasicBlock);
+						}
+					} else {
+						// When we know the index, we can record the value at that index.
+						StoreMethodLocalValue (arrValue.IndexValues, valueToStore.Value, indexToStoreAtInt.Value, curBasicBlock);
```

Comment:
Wondering if we should impose a limit on the number of indices we can model (say, 32?) - there are lots of big initialized arrays in .NET, most of which we don't care about.